src/manifest/manifest_test.ts

```diff
@@ -745,6 +745,32 @@ Deno.test("Manifest() correctly assigns other app features", () => {
   );
 });
 
+Deno.test("Manifest() ensures that apps defining user-level scopes also manage tokens", () => {
+  const definition: SlackManifestType = {
+    runOnSlack: false,
+    name: "fear and loathing in las vegas",
+    description:
+      "fear and loathing in las vegas: a savage journey to the heart of the american dream",
+    displayName: "fear and loathing",
+    icon: "icon.png",
+    botScopes: ["channels:history", "chat:write", "commands"],
+    userScopes: ["users:read"],
+  };
+  const manifest = Manifest(definition);
+
+  // ensure user scopes are set on the manifest
+  assertStrictEquals(
+    manifest.oauth_config.scopes.user,
+    definition.userScopes,
+  );
+
+  // ensure that token management is set to true
+  assertStrictEquals(
+    manifest.oauth_config.token_management_enabled,
+    true,
+  );
+});
+
```

Comment:
Seems like this file defines ` botScopes: []` in many tests 🤔 
Should we also define `userScopes:[]` in those tests as well @filmaj ?

Reply:
I'd like for the tests to use `definition` variables that reflect how end-users would craft their own manifest definitions. In my own experience creating manifest definitions, I would omit the property altogether if I'm not using it. However, devs do the darndest things! I think we should guard against common JS-y/TS-y pitfalls such as what counts as truthy vs. falsy, like empty arrays and empty strings. Therefore, I would recommend the `Manifest()` constructor clean user-provided definitions up into a state that the Slack manifest APIs expect as a baseline.